### PR TITLE
PlainEditor: Add byte selection and navigation operations.

### PR DESCRIPTION
**Pull request by xorgy**

### CODE REVIEW

**Comment by xorgy** [comment]
Splitting out cleanup changes in #147 and #148

**Review by waywardmonkeys — APPROVED** [review]
Basically a rubber stamp as I don't know what exactly each editor operation should be doing and there's no tests.

Here's hoping that Aaron (or someone) can add tests in a future PR!

**Comment by xorgy** [comment]
I promise that if you file an issue, I will write tests at some point for PlainEditor.

**Comment by mwcampbell** [comment]
A bit late, but LGTM.

File: parley/src/layout/editor.rs [hunk 1]
```diff
@@ -111,6 +111,10 @@ where
     BackdeleteWord,
     /// Move the cursor to the cluster boundary nearest this point in the layout.
     MoveToPoint(f32, f32),
+    /// Move the cursor to a byte index.
+    ///
+    /// No-op if index is not a char boundary.
+    MoveToByte(usize),
```
Comment (DJMcNab) [inline parley/src/layout/editor.rs:117]: For all of these, we should document the affinity semantics (in that it clobbers them entirely).

File: parley/src/layout/editor.rs [hunk 2]
```diff
@@ -424,20 +441,7 @@ where
...
-                Affinity::Upstream,
-            )
-        } else {
-            Selection::from_index(
-                &self.layout,
-                new_start.min(self.buffer.len()),
-                Affinity::Downstream,
-            )
-        };
+        self.set_selection(self.cursor_at(start.saturating_add(s.len())).into());
```
Comment (DJMcNab) [inline parley/src/layout/editor.rs:444]: I'm not certain this handles affinity as I'd expect. I'd expect the cursor to be associated with the content you just pasted, not the following cluster.

**Comment by mwcampbell** [comment]
I forgot about affinity semantics. With AccessKit, the client (e.g. screen reader) can theoretically specify the affinity at both ends when setting the selection, though I don't think any of the platform accessibility APIs actually allow that.

**Comment by xorgy** [comment]
@DJMcNab  @mwcampbell Yeah, this is a bit of a mess right now with how `Cursor` currently works, I intend to add some fns to `Cursor` that make it easier to model ‘insertion’ and other behaviors, but the fundamental issue is that Cursors can't be constructed where `index` = `buffer.len()`, which makes all of these behaviors highly conditional.

**Comment by xorgy** [comment]
If we make a `Cursor` type that is based on an insertion index first, like the Parley 0.1 `Cursor`, then all these behaviors become much easier (because we can just always have `Upstream` affinity when inserting, and the neutral empty cursor at `index` = 0 suddenly makes sense)

**Pull request merged**